src/confd/src/infix-dhcp.c

```diff
@@ -22,6 +22,25 @@ static const struct srx_module_requirement reqs[] = {
 	{ NULL }
 };
 
+
+char *first_word(const char *arg, char *buf, size_t len)
+{
+    size_t i = 0, j = 0;
+
+    if (!arg || !buf || len == 0)
+        return NULL;
+
+    while (i < len && isspace(arg[i]))
+        i++;
+
+    while (i < len && !isspace(arg[i]) && j < len - 1)
+        buf[j++] = buf[i++];
+
+    buf[j] = '\0';
+
+    return buf;
+}
+
```

Comment:
Can't we enfore this with a `must` expression?

Reply:
This is for handling option value exceptions, like hostname, other options can have free form text string values, or hex.  It's definitely doable, but it'll take a while to figure out.

Reply:
I got curios :angel:. This seems to work:

```yang
	  must "../name != 'hostname' or re-match(., '[a-zA-Z0-9\\-_]{0,253}')";
```

Reply:
Ha ha ha, OK thank you! I'll push another update :bowing_man::pray: